source_middleware.go

```diff
@@ -139,6 +139,10 @@ type SourceWithSchemaExtraction struct {
 // todo: use https://github.com/ConduitIO/conduit-commons/issues/142 to parse
 // the string value into schema.Type directly.
 func (c *SourceWithSchemaExtraction) SchemaType() schema.Type {
+	if c.SchemaTypeStr == "" {
+		return schema.TypeAvro
```

Comment:
(thinking out loud) Could we initialize it with this value when instantiating `SourceWithSchemaExtraction`?

Reply:
I think that would work if we constraint ourselves to always use a constructor method. It's not terrible, but unless we make the type unexported we can't really force anyone to use the constructor method. I believe that in most cases this should be enough, because when the config is parsed using the `sdk.Util.ParseConfig` function, the default should be set (because we have the `default` tag).